src/BuildInfo.ts

```diff
@@ -24,25 +24,32 @@ export class BuildInfo {
     public static readonly UNKNOWN_VERSION_ID = -1;
     private static readonly MAJOR_VERSION_MULTIPLIER = 10000;
     private static readonly MINOR_VERSION_MULTIPLIER = 100;
-    private static readonly PATTERN = /^([\d]+)\.([\d]+)(?:\.([\d]+))?(-[\w]+)?(-SNAPSHOT)?(-BETA-.)?$/;
 
     public static calculateServerVersionFromString(versionString: string): number {
         if (versionString == null) {
             return BuildInfo.UNKNOWN_VERSION_ID;
         }
-        const info = BuildInfo.PATTERN.exec(versionString);
-        if (info === null) {
+        const mainParts = versionString.split('-');
+        const tokens = mainParts[0].split('.');
+
+        if (tokens.length < 2) {
             return BuildInfo.UNKNOWN_VERSION_ID;
         }
-        const major = Number.parseInt(info[1]);
-        const minor = Number.parseInt(info[2]);
+
+        const major = +tokens[0];
+        const minor = +tokens[1];
         let patch: number;
-        if (info[3] === undefined) {
+        if (tokens.length === 2) {
             patch = 0;
         } else {
-            patch = Number.parseInt(info[3]);
+            patch = +tokens[2];
+        }
+        const version = this.calculateServerVersion(major, minor, patch);
+        if (isNaN(version)) {
```

Comment:
Just wondering, how can this be NaN?

Reply:
If one of major, minor or patch is NaN it will be NaN, because arithmetic operations with NaN results in NaN.

Reply:
Oh, OK. I thought `+"x"` or something like that would throw but it returns `NaN` instead :+1:

Reply:
Exactly.